docs/README.md

```diff
@@ -0,0 +1,148 @@
+# KFServing
+A KFService is a "unit of model serving". Users may have many KFServices, including different definitions across dev, test, and prod environments, as well as across different model use cases. KFServices are defined using the [Kubernetes Custom Resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) standard. 
+
+## Specifying a Service
+### Resource Definition
+This top level resource definition is shared by all Kubernetes Custom Resources.
+
+| Field       |  Value      | Description |
+| ----------- | ----------- | ----------- |
+| kind       | Service                       | https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#types-kinds |
+| apiVersion | serving.kubeflow.org/v1alpha1 | https://kubernetes.io/docs/reference/using-api/api-overview/#api-versioning |
+| metadata   | [Metadata](#Metadata)         | https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata |
+| spec       | [Spec](#Spec)                 | https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status |
+| status     | [Status](#Status)             | https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status |
```

Comment:
Do we need two fields here? seems between label and annotation one can do the job? Or annotation is named more specifically?

Reply:
These are kubernetes standards. At a high level, labels are for identifying resources, and annotations are used for extensibly tuning implementations without mussing with the spec.

See: https://vsupalov.com/kubernetes-labels-annotations-difference/